proto/accessgraph/v1alpha/query.proto

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package accessgraph.v1alpha;
+
+import "teleport/legacy/client/proto/event.proto";
+import "teleport/legacy/types/types.proto";
+
+option go_package = "github.com/gravitational/teleport/gen/proto/go/accessgraph/v1alpha;accessgraphv1alpha";
+
+// AccessGraphService is a service for interacting the access graph service.
+service AccessGraphService {
+  // Query queries the access graph.
+  // Currently only used by WebUI.
+  rpc Query(QueryRequest) returns (QueryResponse);
+  // GetFile gets a static UI file from the access graph container.
+  rpc GetFile(GetFileRequest) returns (GetFileResponse);
+
+  // SendEvent sends an event to the access graph service.
+  rpc SendEvent(SendEventRequest) returns (SendEventResponse);
+
+  // SendResource sends a resource to the access graph service.
+  rpc SendResource(SendResourceRequest) returns (SendResourceResponse);
+}
+
+// Node is a node in the access graph.
+message Node {
+  // ID is the unique ID of the node.
+  string id = 1;
+  // Kind is the kind of the node, ex: "user", "user_group", "resource", etc.
+  string kind = 2;
+  // SubKind is the subkind of the node.
+  string sub_kind = 3;
+  // Name is the name of the node.
+  string name = 4;
+  // Labels is a set of labels for the resource.
+  map<string, string> labels = 5;
+  // Hostname is the hostname of the node.
+  string hostname = 6;
+  // Properties is a set of properties for the resource.
+  map<string, string> properties = 7;
+}
+
+// Edge is an edge in the access graph.
+message Edge {
+  // from is the ID of the node the edge is from.
+  string from = 1;
+  // to is the ID of the node the edge is to.
+  string to = 2;
+  // type is the type of the edge, e.g. "member_of", "belongs_to", etc.
+  string type = 3;
+}
+
+// QueryRequest is a request to query the access graph.
+message QueryRequest {
+  // query is a SQL query.
+  string query = 1;
+}
+
+// QueryResponse is a response to a query.
+message QueryResponse {
+  repeated Node nodes = 1;
+  repeated Edge edges = 2;
+}
+
+// GetFileRequest is a request to get a file.
+message GetFileRequest {
```

Comment:
What file requests are used for?

Reply:
Good question. TAG (the new service that will be using this proto) comes with its UI. This endpoint will be used to serve JS/CSS files from TAG through Teleport Proxy to the user.

Reply:
This could use a lot more documentation, its purpose and what gets returned isn't clear at all.

I'd be also concerned that this RPC could become a dumping group for a large number of features.

Why not static serving, or representing routes in an HTTP server, or an enumeration of the possible files to fetch?